app/recipes_book/views.py

```diff
@@ -0,0 +1,83 @@
+from . import models
+from django.http import JsonResponse, Http404, HttpResponse
+from django.views import View
+import json
+
+
+class RecipesView(View):
+    def get(self, request, recipe_id=None):
+        if recipe_id:
+            recipe = self._load_by_id(recipe_id)
+            return JsonResponse(recipe)
+        elif request.GET.get("name"):
+            filter_name = request.GET.get("name")
+            recipes = self._load_recipes_by_name(filter_name)
+            return JsonResponse(recipes, safe=False)
+        else:
+            recipes = self._load_all_recipes()
+            return JsonResponse(recipes, safe=False)
+
+    def _load_by_id(self, recipe_id):
+        try:
+            recipe = models.Recipe.objects.prefetch_related("ingredients").get(
+                id=recipe_id
+            )
+            return recipe.to_dict()
+        except models.Recipe.DoesNotExist:
+            raise Http404("Recipe not found")
```

Comment:
checkout https://docs.djangoproject.com/en/5.0/topics/http/shortcuts/#get-object-or-404 🙂

Reply:
Tried and it really looks better as you suggested. 👍